drivers/gpio/Kconfig

```diff
@@ -16,6 +16,7 @@ source "subsys/logging/Kconfig.template.log_config"
 
 config GPIO_SHELL
 	bool "GPIO Shell"
+	default y
```

Comment:
Hi Keith, thanks for the review. Sorry for my confusion, are you referring to the PR description above (I've now added) or the next Zephyr release notes?

Reply:
Is defaulting it to `y` necessary?

Reply:
I believe I asked for this originally.
With this change enabling SHELL will automatically enable GPIO_SHELL, rather than requiring both to specified.

This seems to be the default for newer shell interfaces, but there are several older ones which require it explicitly.

Reply:
@keith-zephyr I've updated the v3.6 release notes.

Reply:
There has been further discussion about default behaviour of shells on discord leading to this PR:
https://github.com/zephyrproject-rtos/zephyr/pull/65307

So I suggest removing this change (sorry).